newdle/client/src/components/answer/Calendar.js

```diff
@@ -116,33 +127,55 @@ function getSlotProps(slot, duration, minHour, maxHour, answer) {
   };
 }
 
-function calculateOptionsPositions(options, duration, minHour, maxHour, answers) {
+function calculateOptionsPositions(options, duration, minHour, maxHour, answers, newdleTz, userTz) {
   const optionsByDate = _.groupBy(
-    options.map(slot => getSlotProps(slot, duration, minHour, maxHour, answers[slot])),
-    slot => serializeDate(toMoment(slot.slot, DEFAULT_FORMAT))
+    options.map(slot =>
+      getSlotProps(slot, duration, minHour, maxHour, answers[slot], newdleTz, userTz)
+    ),
+    slot => serializeDate(slot.groupDateKey, HTML5_FMT.DATE, userTz)
   );
 
   return Object.entries(optionsByDate).map(([date, options]) => {
     return {date, optionGroups: groupOverlaps(options, duration)};
   });
 }
 
-function getBusySlotProps(slot, minHour, maxHour) {
+function getBusySlotProps(slot, minHour, maxHour, date, newdleTz, userTz) {
   const [startTime, endTime] = slot;
-  const start = toMoment(startTime, 'HH:mm');
-  const end = toMoment(endTime, 'HH:mm');
+  const start = toMoment(startTime, 'HH:mm', newdleTz).tz(userTz);
+  const end = toMoment(endTime, 'HH:mm', newdleTz).tz(userTz);
+
+  const startDatetime = toMoment([date, startTime].join('T'), DEFAULT_FORMAT, newdleTz).tz(userTz);
+  const pos = calculatePosition(startDatetime, minHour, maxHour);
+  const startTimeLocal = serializeDate(startDatetime, 'HH:mm', userTz);
+  const endTimeLocal = serializeDate(end, 'HH:mm', userTz);
+
   return {
     startTime,
     endTime,
+    startTimeLocal,
+    endTimeLocal,
+    groupDateKey: startDatetime,
     height: calculateHeight(start, end, minHour, maxHour),
-    pos: calculatePosition(start, minHour, maxHour),
-    key: `${startTime}-${endTime}`,
+    pos,
+    key: `${startTimeLocal}-${endTimeLocal}`,
   };
 }
 
-function calculateBusyPositions(busyTimes, minHour, maxHour) {
-  return Object.entries(busyTimes).map(([date, times]) => {
-    return {date, times: times.map(slot => getBusySlotProps(slot, minHour, maxHour))};
+function calculateBusyPositions(busyTimes, minHour, maxHour, newdleTz, userTz) {
+  const busySlots = [];
+  Object.entries(busyTimes).forEach(([date, times]) => {
+    busySlots.push(
+      ...times.map(slot => getBusySlotProps(slot, minHour, maxHour, date, newdleTz, userTz))
+    );
+  });
+
+  const busyByDate = _.groupBy(busySlots, slot =>
+    serializeDate(slot.groupDateKey, HTML5_FMT.DATE, userTz)
+  );
+
+  return Object.entries(busyByDate).map(([date, times]) => {
+    return {date, times};
   });
```

Comment:
I think you don't need any of the changes here if you make this change instead:

```patch
diff --git a/newdle/client/src/components/answer/AnswerPage.js b/newdle/client/src/components/answer/AnswerPage.js
index c25342d..87b7212 100644
--- a/newdle/client/src/components/answer/AnswerPage.js
+++ b/newdle/client/src/components/answer/AnswerPage.js
@@ -150,9 +150,9 @@ export default function AnswerPage() {

   useEffect(() => {
     if ((participantCode && !participantUnknown) || (!participantCode && user)) {
-      dispatch(fetchBusyTimesForAnswer(newdleCode, participantCode || null, dates, newdleTz));
+      dispatch(fetchBusyTimesForAnswer(newdleCode, participantCode || null, dates, userTz));
     }
-  }, [dates, newdleCode, participantCode, participantUnknown, user, newdleTz, dispatch]);
+  }, [dates, newdleCode, participantCode, participantUnknown, user, userTz, dispatch]);

   if (!newdle || (participantCode && !participant)) {
     return null;
```

(you probably need to still change some params or undo changes related to the new props you added )

Reply:
You were totally right, so much code simplified, thanks for that!